src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -10,29 +9,23 @@ internal sealed class ImmutableDefinitionChecker {
 
 		internal delegate void DiagnosticSink( Diagnostic diagnostic );
 
-		private readonly SemanticModel m_model;
+		private readonly Compilation m_compilation;
 		private readonly DiagnosticSink m_diagnosticSink;
 		private readonly ImmutabilityContext m_context;
 
 		public ImmutableDefinitionChecker(
-			SemanticModel model,
+			Compilation compilation,
 			DiagnosticSink diagnosticSink,
 			ImmutabilityContext context
 		) {
-			m_model = model;
+			m_compilation = compilation;
 			m_diagnosticSink = diagnosticSink;
 			m_context = context;
 		}
 
-		public bool CheckStruct( StructDeclarationSyntax decl ) => CheckDeclaration( decl );
-		public bool CheckClass( ClassDeclarationSyntax decl ) => CheckDeclaration( decl );
-
-		// This method is private to ensure callers only pass structs & classes
-		private bool CheckDeclaration( TypeDeclarationSyntax decl ) {
+		public bool CheckDeclaration( ITypeSymbol type ) {

```

Comment:
consider `INamedTypeSymbol`? `GetDeclaredSymbol` of a `TypeDeclarationSyntax` returns an `INameTypeSymbol` anyway.